pynamodb/attributes.py

```diff
@@ -352,6 +381,49 @@ def __ne__(self, other: Any) -> bool:
         return self is not other
 
 
+class DiscriminatorAttribute(Attribute[type]):
+    attr_type = STRING
+
+    def __init__(self, attr_name: Optional[str] = None) -> None:
+        super().__init__(attr_name=attr_name)
+        self._class_map: Dict[type, Any] = {}
+        self._discriminator_map: Dict[Any, type] = {}
+
+    def register_class(self, cls: type, discriminator: Any):
```

Comment:
```suggestion
    def register_class(self, cls: type, discriminator: Union[Callable[..., Any], str]):
```

Reply:
Is there a desire to allow other types (say numbers) in here?

Reply:
yeah i kept this as any so that a user could subclass as needed with minimal effort

Reply:
cool, seems reasonable